python/lsst/ts/observatory/control/maintel/mtcs.py

```diff
@@ -822,6 +830,100 @@ async def _wait_for_mtm1m3_detailed_state(
                 )
             self.log.debug(f"M1M3 detailed state {m1m3_detailed_state.detailedState!r}")
 
+    async def _wait_hard_point_test_ok(self, hp: int) -> None:
+        """Wait until the hard point test for the specified hard point
+        finishes.
+
+        Parameters
+        ----------
+        hp : `int`
+            Index of the hard point (starting from 1).
+
+        Raises
+        ------
+        RuntimeError
+            If the hp test failed.
+        """
+
+        self.log.info("Checking if the hard point breakaway test has passed.")
+
+        while True:
+            hp_test_state = MTM1M3.HardpointTest(
+                (
+                    await self.rem.mtm1m3.evt_hardpointTestStatus.next(
+                        flush=False, timeout=self.long_timeout
+                    )
+                ).testState[hp - 1]
+            )
+
+            if hp_test_state == MTM1M3.HardpointTest.FAILED:
+                raise RuntimeError(f"Hard point {hp} test FAILED.")
+            elif hp_test_state == MTM1M3.HardpointTest.PASSED:
+                self.log.info(f"Hard point {hp} test PASSED.")
+                return
+            else:
+                self.log.info(f"Hard point {hp} test state: {hp_test_state!r}.")
+
+    async def _wait_bump_test_ok(
+        self, actuator_id: int, primary: bool, secondary: bool
+    ) -> None:
+        """Wait until the bump test for the specified actuator finishes.
+
+        Parameters
+        ----------
+        actuator_id : `int`
+            Actuator id.
+        primary : `bool`
+            Wait for primary (z-axis) test to finish.
+        secondary : `bool`
+            Wait for secondary (xy-axis) test to finish.
+        """
+
+        actuator_index = self.get_m1m3_actuator_index(actuator_id)
+
+        while True:
+            bump_test_status = (
+                await self.rem.mtm1m3.evt_forceActuatorBumpTestStatus.next(
+                    flush=False, timeout=self.long_timeout
+                )
+            )
+
+            if actuator_id != bump_test_status.actuatorId:
+                raise RuntimeError(
+                    f"Expecting status from actuator {actuator_id} got {bump_test_status.actuatorId}."
+                )
+
+            primary_status = MTM1M3.BumpTest(
+                bump_test_status.primaryTest[actuator_index]
+            )
+            secondary_status = MTM1M3.BumpTest(
+                bump_test_status.secondaryTest[actuator_index]
```

Comment:
secondaryTest size is 112. This is indexed by so called secondary index - FATABLE_SINDEX. You need to retrieve that from FATABLE, and use that to get secondary status